src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -151,25 +157,28 @@ private void OnOutputDataReceived(object sender, DataReceivedEventArgs e)
 
         private void OnProcessExited(object sender, EventArgs e)
         {
-            string exceptionMessage = string.Join(",", _processStdErrDataQueue.Where(s => !string.IsNullOrEmpty(s)));
-            try
+            if (!_disposing)
```

Comment:
OnProcessExited event was getting triggered during jobhost restarts multiple times as we explicitly kill the process